dpgen/ops/dispatcherop.py

```diff
@@ -0,0 +1,118 @@
+from abc import abstractmethod
+from pathlib import Path
+from typing import List
+from itertools import chain
+
+from dpdispatcher import Machine, Resources, Task, Submission
+
+from .op import OP, Status
+from .opio import OPIO
+from .context import IterationContext, task_pattern
+
+
+class DispatcherOP(OP):
+    r"""This OP will call DPDispatcher to execuate commands on remote.
+    
+    Parameters
+    ----------
+    context : IterationContext
+        DP-GEN iteration context
+    machine : dict
+        DPDispatcher machine parameters. See also DPDispatcher doc.
+    resources : dict
+        DPDispatcher machine parameters. See also DPDispatcher doc.
+    command : str
+        path or command to a certain program on remote
+    """
+    @OP.set_status(status = Status.INITED)
+    def __init__(
+            self,
+            context: IterationContext,
+            machine: dict,
+            resources: dict,
+            command: str,
+    )->None:
+        super().__init__(context)
+        self.machine = Machine(**machine)
+        self.resources = Resources(**resources)
+        self.user_command = command
+    
+    @property
+    def command(self) -> str:
+        """str: The full bash command produced by the command the user provides."""
+        return self.user_command
+
+    @property
+    def task_pattern(self) -> str:
+        """str: task pateern for glob, e.g. `task.*`"""
+        return task_pattern
+    
+    @property
+    def outlog(self) -> str:
+        return "log"
+
+    @property
+    def errlog(self) -> str:
+        return "log"
+    
+    @property
+    @abstractmethod
+    def forward_files(self) -> List[str]:
+        """List[str]: relative path to task path"""
+
+    @property
+    @abstractmethod
+    def backward_files(self) -> List[str]:
+        """List[str]: relative path to task path"""
+    
+    @property
+    def forward_common_files(self) -> List[str]:
+        """List[str]: relative path to work path"""
+        return []
+
+    @property
+    def backward_common_files(self) -> List[str]:
+        return []
+    
+    @property
+    def task_paths(self) -> List[Path]:
+        if self.status is not Status.EXECUTED:
+            raise RuntimeError
+        return self.work_path.glob(self.task_pattern)
```

Comment:
We may not want to use pattern to find tasks, but accept a set of Path objects. For example the "train_dirs" as output of `PrepDPTrain`

https://github.com/amcadmus/dpgen/blob/05a74613449ae67867328d1f67d949a0a99fe45f/dpgen/ops/prep_dp_train.py#L122-L127

Reply:
I haven't understood it...

Reply:
I suggest that we d not use task_pattern, but explicitly pass the Set of tasks.

Reply:
`self.numb_models` is set by users, but, do users know really know how many tasks there will be before running the OP?

Reply:
Yes, the working path of training tasks are output by `PrepDPTrain`, thus the dpgen knows which training task to run.